inst/include/mode/r/mode.hpp

```diff
@@ -54,5 +54,22 @@ cpp11::sexp mode_solve(SEXP ptr, double end_time) {
   return state_array(dat, obj->n_state(), obj->n_particles());
 }
 
+double validate_time(cpp11::sexp r_time) {
+  cpp11::doubles time = cpp11::as_cpp<cpp11::doubles>(r_time);
+  if (time.size() > 1) {
+    cpp11::stop("time should be a scalar value");
```

Comment:
avoid should where it's an error ("expected time to be a scalar", or "time must be a scalar") - I know this is pedantic but there's a [usage guide](https://www.ietf.org/rfc/rfc2119.txt) :)

Typically we've been single quoting `time` in messages like this (as it reflects an argument name to the user).

Reply:
not pedantic, thanks for pointing out :)